pre_commit/error_handler.py

```diff
@@ -63,9 +68,9 @@ def error_handler() -> Generator[None, None, None]:
         yield
     except (Exception, KeyboardInterrupt) as e:
         if isinstance(e, FatalError):
-            msg = 'An error has occurred'
+            msg, ret_code = 'An error has occurred', 1
         elif isinstance(e, KeyboardInterrupt):
-            msg = 'Interrupted (^C)'
+            msg, ret_code = 'Interrupted (^C)', 130
```

Comment:
on python 3.8 it will set the exit code correctly if you let it raise

Reply:
that would produce a stacktrace so no please don't do this

Reply:
@graingert I completely understand your point (I also would like to have the appropriate return code for Windows too), but It would be a more complicated PR in that case. Maybe the Windows support should be a separate effort.